wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -217,159 +146,181 @@ class PresentationActivity : FragmentActivity() {
+                                val document = findSuitableDocumentForRequest(docRequest)
+                                // if document == null no suitable documents could be found then skip to next docRequest
+                                val mdocCredential = document.findCredential(
+                                    WalletApplication.CREDENTIAL_DOMAIN_MDOC,
+                                    Clock.System.now()
+                                ) as MdocCredential
+
+                                // extract the TrustPoint if possible
+                                var trustPoint: TrustPoint? = null
+                                if (docRequest.readerAuthenticated) {
+                                    val result = walletApp.trustManager.verify(
+                                        docRequest.readerCertificateChain!!.javaX509Certificates,
+                                        customValidators = emptyList()  // not needed for reader auth
                                     )
+                                    if (result.isTrusted && result.trustPoints.isNotEmpty()) {
+                                        trustPoint = result.trustPoints.first()
+                                    } else if (result.error != null) {
+                                        Logger.w(
+                                            com.android.identity_credential.wallet.presentation.TAG,
+                                            "Error finding TrustPoint for reader auth",
+                                            result.error!!
+                                        )
+                                    }
                                 }
-                        }
-
-                        State.RESPONSE_SENT -> {
-                            stateDisplay.value = "Response Sent"
-                            Logger.i(TAG, "State: Response Sent")
-                        }
 
-                        else -> {}
-                    }
-                }
+                                // generate the DocumentRequest from the current docRequest
+                                val documentRequest = MdocUtil.generateDocumentRequest(docRequest)
+
+                                // show the Presentation Flow for and get the response bytes for
+                                // the generated Document
+                                val responseBytes = showPresentmentFlow(
+                                    activity = this@PresentationActivity,
+                                    walletApp = walletApp,
+                                    documentRequest = documentRequest,
+                                    mdocCredential = mdocCredential,
+                                    trustPoint = trustPoint,
+                                    encodedSessionTranscript = deviceRetrievalHelper!!.sessionTranscript
+                                )
+                                sendResponseToDevice(responseBytes)
```

Comment:
This won't work with multi-document presentations b/c here we're sending a `DeviceResponse` per document request. This is easily fixed by changing `showPresentmentFlow()` so it returns the bytes of `Document` CBOR according to 18013-5. That is, the output of `DocumentGenerator.generate()`.

With this, you simply init a `DeviceResponseGenerator` at the top (before `deviceRequest.docRequests.forEach` loop), then call `addDocument(documentResponse)` and after the loop, you can do `deviceRetrievalHelper.sendDeviceResponse(deviceResponseGenerator.generate())`

Reply:
ah I see, I literally had this exact code that you mention in the first commit but it was in the wrong layer, I'll bring it back on `PresentationActivity` 🙏